include/posix/mqueue.h

```diff
@@ -17,7 +17,10 @@ extern "C" {
 #endif
 
 typedef void *mqd_t;
+#ifndef _MODE_T_DECLARED
 typedef unsigned int mode_t;
```

Comment:
Well, it's implausible situation that mode_t is defined by mqueue.h. Per POSIX, it's defined by sys/types.h: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html . So, let's start to actually fix these, not work around. I.e., should be added to lib/libc/minimal/include/sys/types.h for minimal libc case, it's of course already there for newlib, and should be removed from include/posix/*.

Reply:
What's interesting is we also define `mode_t` in `include/posix/unistd.h`

Reply:
All POSIX headers is complete mess. And we either clean them up and rely on newlib headers, or choose the hard way of cleaning them up **and** recreating newlib headers for minimal libc ;-).

Reply:
We'll I cleaned it up so mode_t comes from newlib or minlibc types.h